lib/mapping.ts

```diff
@@ -0,0 +1,302 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+'use strict'
+
+import { cloneDeep } from 'lodash'
+import { Schema } from 'mongoose'
+import { MongoosasticDocument, MongoosasticModel } from 'types'
+
+//
+// Get type from the mongoose schema
+//
+// Returns the type, so in case none is set, it's the mongoose type.
+//
+// @param paths
+// @param field
+// @return the type or false
+
+function getTypeFromPaths (paths: Record<string, any>, field: string) {
+	let type = false
+
+	if (paths[field] && paths[field].options.type === Date) {
+		return 'date'
+	}
+
+	if (paths[field] && paths[field].options.type === Boolean) {
+		return 'boolean'
+	}
+
+	if (paths[field]) {
+		type = paths[field].instance ? paths[field].instance.toLowerCase() : 'object'
+	}
+
+	return type
+}
+
+//
+// Generates the mapping
+//
+// Can be called recursively.
+//
+// @param cleanTree
+// @param inPrefix
+// @return the mapping
+//
+function getMapping (cleanTree: Record<string, any>, inPrefix: string) {
+	const mapping: Record<string, any> = {}
+	let value
+	let field: string
```

Comment:
this declarations are not needed / I have removed this in the refactoring i did, they are declared and used inside the loops down

Reply:
We can open a PR to refactor this one.